### PR TITLE
Update test_cli.py

### PR DESCRIPTION
**Pull request by jseagrave21**
**What current issue(s) does this address, or what feature is it adding?**
This PR attempts to address issue #47 

**How did you solve this problem?**
Introduced a new test for main()

**How did you make sure your solution works?**
tested it

**Did you add any tests?**
yes

**Did you run `make lint` and `make coverage`?**
yes. What is odd to me though, is that coverage on my instance did not change.

**Are there any special changes in the code that we should be aware of?**
My new version of the test makes use of subprocess and warning modules

### CODE REVIEW

**Comment by metachris** [comment]
The error is here: https://travis-ci.org/CityOfZion/neo-python-core/builds/415061740#L773

There is no `np-utils` binary available, because we don't do `pip install -e .`.  Without this, only the dependencies are installed,  but not the neocore module itself.

You could change these lines: https://github.com/CityOfZion/neo-python-core/blob/master/.travis.yml#L14 to

    - pip install -r requirements_dev.txt
    - pip install -e .
    - pip install coveralls

**Comment by jseagrave21** [comment]
@metachris Awesome! Thank you!

**Comment by metachris** [comment]
Actually, I think we could perhaps exclude `cli.py` from the coverage count, and just test the functions within. Not sure, but might be a way.

**Comment by metachris** [comment]
I think some of the latest code changes broke this test. Should have been this: #58

**Comment by jseagrave21** [comment]
@metachris Updated now.

File: tests/test_cli.py [hunk 1]
```diff
@@ -32,3 +33,24 @@ def test_create_wallet(self):
         self.assertIn("address", wallet)
         self.assertIsInstance(wallet["private_key"], str)
         self.assertIsInstance(wallet["address"], str)
+
+    def test_main(self):
+        warnings.filterwarnings('ignore', category=ResourceWarning)  # filters warnings about subprocess files still being open
+
+        version = subprocess.Popen(['np-utils', '--version'], stdout=subprocess.PIPE)
+        self.assertEqual(version.stdout.read(), b'neocore v0.5.2-dev\n')
```
Comment (ixje) [inline tests/test_cli.py]: The problem with this hardcoded approach is that with each release we need to update this code. I think a workaround is to read the version from the init file 
```
import neocore
print(neocore.__version__)
```

Comment (jseagrave21) [reply tests/test_cli.py]: 👍

**Pull request merged**